post_info.php

```diff
@@ -0,0 +1,688 @@
+    echo json_encode($return_json);
+    exit();
+}
+
+/**
+ * helper function for executing the SQL statement of updating courses/sections
+ * for one specific term
+ * @param association_list:array a list of user associations
+ * @param action:string the actions that the user wants to perform
+ */
+function handle_courses_sections($association_list, $action)
+{
+    // Initialize a success status array
+    $return_data = array();
+    // go through each user association in the list and make seperate SQL calls
+    for ($i = 0; $i < count($association_list); $i++) {
+        $course = $association_list[$i]->course;
+        $section = $association_list[$i]->section;
+        $user_id = $association_list[$i]->user_id;
+        $course_code = $course->course_code;
+        $course_title = $course->title;
+        $department_name = $course->department_name;
+        $term = $course->term;
+        // Initialize $bind_variables
+        $bind_variables = array();
+        if ($action == "delete") {
+            if ($section != null) {
+                $bind_variables["section_id"] = $section->section_id;
+            } else {
+                $bind_variables["course_code"] = $course_code;
+            }
+        } elseif ($action == "add_or_update") {
+            $bind_variables["course_code"] = $course_code;
+            $bind_variables["course_title"] = $course_title;
+            $bind_variables["department_name"] = $department_name;
+            $bind_variables["section_code"] = $section->section_code;
+            $bind_variables["term"] = $term;
+            if ($section->section_id != null) {
+                $bind_variables["section_id"] = $section->section_id;
+            }
+        }
+        // generate the SQL statement using the information provided
+        $sql = gen_query_update_courses_sections($action, $section);
+        $status = execute_sql($sql, $bind_variables, null);
+        // Determine the status and then add the status object in the returned array
+        if ($status == "success!") {
+            $temp = array('TYPE' => 'success', 'data' => null);
+            array_push($return_data, $temp);
+        } else {
+            $temp = array();
+            $temp["TYPE"] = "error";
+            $temp["data"] = $status;
+            array_push($return_data, $temp);
+        }
+    }
+    // Encapsulate the data into a php object
+    $return_json = array();
+    $return_json["TYPE"] = "success";
+    $return_json["data"] = $return_data;
+    echo json_encode($return_json);
+    exit();
+}
+
+/**
+ * helper function for executing the SQL statement of updating survey settings
+ * @param survey_id:int The id of the survey
+ * @param level:string dept, course, section
+ * @param user_id:string The utorid of the user
+ * @param action:string add_or_update, delete, branch
+ * @param data:object the body of the request
+ */
+function handle_survey_setting($survey_id, $level, $user_id, $action, $data)
+{
```

Comment:
Since "handle_survey_setting" is too long, I plan to write helper functions to modularize the logic.

Reply:
fixed